### ISSUE TITLE
watchTermination not report exception

### ISSUE THREAD

**Issue by lfygh** [akka/akka#31904]
the [demo code](https://doc.akka.io/docs/akka/current/stream/operators/Source-or-Flow/watchTermination.html) does not work as expected 
```
Source.range(1, 5)
    .watchTermination(
        (prevMatValue, completionStage) -> {
          // this function will be run when the stream terminates
          // the CompletionStage provided as a second parameter indicates whether
          // the stream completed successfully or failed
          completionStage.whenComplete(
              (done, exc) -> {
                if (done != null)
                  System.out.println("The stream materialized " + prevMatValue.toString());
                else System.out.println(exc.getMessage());
              });
          return prevMatValue;
        })
    .runForeach(
        element -> {
          if (element == 3) throw new Exception("Boom");
          else System.out.println(element);
        },
        system);
/*
Prints:
1
2
Boom
 */
```
expect print error msg, but it print "The stream materialized " + prevMatValue.toString()

version: akka-stream_2.13-2.5.32-sources.jar

**Comment by lfygh** [comment]
1、

> ```scala
> //>using dep "com.typesafe.akka::akka-stream:2.8.0"
> 
> import akka.actor.ActorSystem
> import akka.stream.scaladsl.Source
> 
> import scala.concurrent.ExecutionContext
> import scala.util.{ Failure, Success }
> 
> 
>     implicit val system: ActorSystem = ActorSystem("akka")
>     implicit val ec: ExecutionContext = system.dispatcher
> 
> Source(1 to 5)
>   .watchTermination()(
>     (prevMatValue, future) =>
>       // this function will be run when the stream terminates
>       // the Future provided as a second parameter indicates whether the stream completed successfully or failed
>       future.onComplete {
>         case Failure(exception) => println(exception.getMessage)
>         case Success(_)         => println(s"The stream materialized $prevMatValue")
>       })
>   .runForeach(println)
> ```
> 
> and it gives me ![image](https://user-images.githubusercontent.com/501740/231246942-5df5efe5-35a8-43d2-9522-556ce3e9888e.png)

1、Your demo is not the same as mine.

2、Yes, the bug is fixed at 'com.typesafe.akka::akka-stream:2.8.0'

**Comment by He-Pin** [comment]
![image](https://user-images.githubusercontent.com/501740/231417990-c1c6f195-5525-4905-bac0-c5716a3d7089.png)

@lfygh it was fixed in 2.6.20 too
